internal/pkg/template/diff/diff.go

```diff
@@ -88,7 +89,80 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	toSeq, fromSeq := make([]yaml.Node, len(to.Content)), make([]yaml.Node, len(from.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	seen, err := compare(fromSeq, toSeq)
+	if err != nil {
+		return nil, err
+	}
+	var index int
+	children := make(map[string]*Node)
+	for _, occurrence := range seen {
+		if occurrence.inTo == occurrence.inFrom {
+			continue
+		}
+		for i := 0; i < occurrence.inFrom-occurrence.inTo; i++ {
+			children[strconv.Itoa(index)] = &Node{
+				key:      strconv.Itoa(index),
+				oldValue: &(occurrence.node),
+			}
+			index++
+		}
+		for i := 0; i < occurrence.inTo-occurrence.inFrom; i++ {
+			children[strconv.Itoa(index)] = &Node{
+				key:      strconv.Itoa(index),
+				newValue: &(occurrence.node),
+			}
+			index++
+		}
+	}
+	return children, nil
+}
+
+type nodeOccurrence struct {
+	node   yaml.Node
+	inTo   int
+	inFrom int
+}
+
+func compare(from, to []yaml.Node) (map[string]*nodeOccurrence, error) {
```

Comment:
What do you think of renaming the function and changing the return type as follows:
```suggestion
// nodeCounter holds the number of times a yaml.Node appears in a sequence.
type nodeCounter struct {
    node yaml.Node
    counts []int // Number of times node appeared in each sequence.
}

// countDistinct returns a slice of each unique yaml.Node across sequences and the number of times the node appeared in each sequence.
func countDistinct(sequences ...[]yaml.Node) ([]*nodeCounter, error) {
```

My concerns were that `compare` is a bit too generic as a name for what this function is doing (it's really just counting).
Similarly, I wasn't sure if a `map[string]` is needed because clients don't care about the key (the serialized yaml.Node) 

Let me know what you think!